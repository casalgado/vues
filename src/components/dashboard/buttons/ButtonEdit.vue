<template>
  <b-button @click="edit">edit</b-button>
</template>
<script>
import { getById } from "@/firebase";
import { mapState } from "vuex";

export default {
  name: "ButtonPaid",
  props: {
    oid: String,
    path: String
  },
  data() {
    return {
      object: {}
    };
  },
  computed: {
    ...mapState(["ref"]),
    paid() {
      return this.object.paid != "";
    }
  },
  methods: {
    edit: function() {
      console.log(this.object);
      this.$router.push({
        name: "EditOrder",
        params: { object: this.object }
      });
    }
  },
  mounted() {
    getById(`${this.ref}/${this.path}/${this.oid}`).then(e => {
      e.id = this.oid;
      this.object = e;
    });
  }
};
</script>
<style scoped>
</style>