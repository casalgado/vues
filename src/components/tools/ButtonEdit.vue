<template>
  <b-button @click="edit" variant="primary">edit</b-button>
</template>
<script>
export default {
  name: "ButtonEdit",
  props: {
    oid: String,
    destination: String,
  },
  methods: {
    edit: function() {
      this.$router.push({
        name: "EditOrder",
        params: { oid: this.oid },
      });
    },
  },
  mounted() {},
};
</script>
<style scoped></style>
