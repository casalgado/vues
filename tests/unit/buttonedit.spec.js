import { mount, createLocalVue } from "@vue/test-utils";
import ButtonEdit from "../../src/components/tools/ButtonEdit";
import BootstrapVue from "bootstrap-vue";

const localVue = createLocalVue();
localVue.use(BootstrapVue);

const $router = [];

describe("Component", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(ButtonEdit, { localVue });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders correctly", () => {
    const wrapper = mount(ButtonEdit, { localVue });
    expect(wrapper.element).toMatchSnapshot();
  });

  test("calls router when edit method is called", () => {
    const wrapper = mount(ButtonEdit, {
      localVue,
      mocks: { $router },
      propsData: { oid: "123", destination: "path/to/destination" },
    });
    wrapper.vm.edit();
    expect(wrapper.vm.$router[0].params.oid).toEqual("123");
    expect(wrapper.vm.$router[0].name).toEqual("path/to/destination");
  });
});
