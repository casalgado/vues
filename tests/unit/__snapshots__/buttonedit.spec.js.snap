// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<button
  class="btn toolbox-button btn-primary"
  type="button"
>
  editar
</button>
`;
